lua/orgmode/org/mappings.lua

```diff
@@ -218,37 +218,78 @@ function OrgMappings:handle_return(suffix)
   suffix = suffix or ''
   local item = Files.get_current_file():get_current_node()
   if item.type == 'headline' or item.node:parent():type() == 'headline' then
-    if item.type ~= 'headline' then
+    if item.type ~= 'stars' then
```

Comment:
Why is this changed? `stars`, `item` and `tags` nodes are all on the same level, so this should handle return from all of those.
Also, below `section` variable will not return section node if you are on `item` or `tags` node, since this will go one parent up, and that will go 2 more parents up, which can end up on the top level document. Maybe I'm missing something, didn't tested it yet.

Reply:
The logic before was a bit broken, but still worked.
If it was not a `headline` type, it does `ts_utils.goto_node(item.node:parent())`, which goes to the start of the headline. But then `Files.get_current_file():get_current_node()` returns the smallest node at that point, which is `stars`.
But that's fine, as stars also has the level.
But now for the change I made, I need to know which level I am on, so I can go up to the section. So if the cursor is between the stars and the line, it is type `headline`. In that case I do the same as before to change `item` to `stars`.

I agree, it is not really obvious. Maybe the code should just walk up the parent until it reaches `section` to be easier to understand.

Reply:
> I agree, it is not really obvious. Maybe the code should just walk up the parent until it reaches section to be easier to understand.


I like this. You can introduce a helper in utils that would do this traversing to closest parent, something like good ol' Jquery's [parentsUntil](https://api.jquery.com/parentsuntil/)